rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -209,6 +210,23 @@ object SystemProcesses {
           produce(Seq(Par()), ack)
       }
 
+      def registryOps: Contract[F] = {
+        case isContractCall(
+            produce,
+            Seq(RhoType.String("buildUri"), argument, ack)
+            ) =>
+          val response = argument match {
+            case RhoType.Name(gprivate) =>
+              val hashKeyBytes = Blake2b256.hash(gprivate.id.toByteArray)
+              RhoType.Uri(Registry.buildURI(hashKeyBytes))
+            case RhoType.ByteArray(ba) =>
+              val hashKeyBytes = Blake2b256.hash(ba)
+              RhoType.Uri(Registry.buildURI(hashKeyBytes))
+            case _ => Par()
+          }
+          produce(Seq(response), ack)
+      }
```

Comment:
Please add RhoSpec for this contract, including one showing that

`buildUri!(name) == buildUri!(name.toByteArray)`

If the above is not true, let's just have one overload of `buildUri`, taking a byteArray.

Reply:
Done